src/frontend/src/optimizer/plan_node/logical_agg.rs

```diff
@@ -384,6 +421,17 @@ impl LogicalAgg {
             .any(|call| matches!(call.agg_kind, AggKind::StringAgg | AggKind::ArrayAgg))
     }
 
+    pub(crate) fn can_agg_two_phase(&self) -> bool {
+        self.agg_calls().iter().all(|call| {
+            matches!(
+                call.agg_kind,
+                AggKind::Min | AggKind::Max | AggKind::Sum | AggKind::Count
+            ) && !call.distinct
+            // TODO: why do we need `&& call.order_by_fields.is_empty()` ?
```

Comment:
Seems this check is unnecessary for now.

Reply:
> Seems this check is unnecessary for now.

ok. I've commented this.